qiskit/algorithms/evolvers/variational/calculators/evolution_grad_calculator.py

```diff
@@ -0,0 +1,106 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Calculates and binds evolution gradients for Variational Quantum Time Evolution."""
+
+from typing import Union, Optional, List, Dict, Callable
+
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import ParameterVector, ParameterExpression, Parameter
+from qiskit.opflow import StateFn, Gradient, CircuitGradient, OperatorBase, Z, CircuitSampler
+from qiskit.opflow.gradients.circuit_gradients import LinComb
+
+
+def calculate(
+    observable: OperatorBase,
+    ansatz: Union[StateFn, QuantumCircuit],
+    parameters: Optional[Union[ParameterVector, ParameterExpression, List[ParameterExpression]]],
+    grad_method: Union[str, CircuitGradient],
+    basis: OperatorBase = Z,
+) -> OperatorBase:
+    """
+    Calculates a parametrized evolution gradient object.
+
+    Args:
+        observable: Observable for which an evolution gradient should be calculated,
+            e.g., a Hamiltonian of a system.
+        ansatz: Quantum state to be evolved in the form of a parametrized quantum circuit.
+        parameters: Parameters with respect to which gradients of an ansatz should be computed.
+        grad_method: The method used to compute the state gradient. Can be either
+            ``'param_shift'`` or ``'lin_comb'`` or ``'fin_diff'``.
+        basis: Basis with respect to which evolution gradient is calculated. In case of a default
+            Z basis a real part of an evolution gradient is returned. In case of a Y basis,
+            an imaginary part of an evolution gradient is returned.
+
+    Returns:
+        Parametrized evolution gradient as an OperatorBase.
+    Raises:
+        ValueError: If unsupported value for ``basis`` is provided for a given gradient method.
+    """
+    operator = StateFn(observable, is_measurement=True) @ StateFn(ansatz)
+    if grad_method == "lin_comb":
+        return LinComb(aux_meas_op=basis).convert(operator, parameters)
+    elif basis != Z:
+        raise ValueError(
+            f"Basis which is not Z is only supported for ``lin_comb`` method. Provided method is "
+            f"{grad_method} with basis {basis}."
+        )
+    return Gradient(grad_method).convert(operator, parameters)
```

Comment:
With the move to primitives we very likely won't have opflow objects for gradients/QFI anymore that one still has to call `eval` on. But rather you just have one object which gives you the result directly. So to avoid changing this interface in the future, it might be better to have this method evaluate the gradients directly? (So basically integrating the `eval_grad_result` from below)